java/ruleXtract/src/it/scala/uk/ac/cam/eng/PipelineTest.scala

```diff
@@ -0,0 +1,188 @@
+import uk.ac.cam.eng.extraction.hadoop.datatypes.FeatureMap
+import uk.ac.cam.eng.extraction.hadoop.datatypes.ExtractedData
+
+class PipelineTest extends FunSuite with BeforeAndAfterAll {
+
+  override def beforeAll(configMap: ConfigMap) {
+    RuleExtractorTest.folder.create()
+    RuleExtractorTest.setupFileSystem()
+  }
+
+  override def afterAll(configMap: ConfigMap) {
+    RuleExtractorTest.folder.delete()
+  }
+
+  def filterSequenceFile(input: Path, output: Path, conf: Configuration) = {
+    val reader = new SequenceFile.Reader(FileSystem.get(conf), input, conf);
+    val key = ReflectionUtils.newInstance(
+      reader.getKeyClass(), conf).asInstanceOf[Writable]
+    val value = ReflectionUtils.newInstance(
+      reader.getValueClass(), conf).asInstanceOf[Writable]
+    for (writer <- 
+        managed(new SequenceFile.Writer(FileSystem.get(conf), conf, output, key.getClass, value.getClass))) {
+      var count = 0
+      while (reader.next(key, value)) {
+        if (count % 10 == 0)
+          writer.append(key, value)
+        count += 1
+      }
+    }
+  }
+
+  def assertWithDelta(expected: Double)(result: Double) = {
+    val delta = 1D / 1024D
+    assert(Math.abs(expected - result) < delta)
+  }
+
+  test("The rule extraction job") {
+    val conf = new Configuration
+    conf.set("mapreduce.framework.name", "local");
+    conf.setInt(CLI.RuleParameters.MAX_SOURCE_PHRASE, 9)
+    conf.setInt(CLI.RuleParameters.MAX_SOURCE_ELEMENTS, 5)
+    conf.setInt(CLI.RuleParameters.MAX_TERMINAL_LENGTH, 5)
+    conf.setInt(CLI.RuleParameters.MAX_NONTERMINAL_SPAN, 10)
+    conf.setBoolean(CLI.ExtractorJobParameters.REMOVE_MONOTONIC_REPEATS, true)
+    conf.setBoolean(CLI.ExtractorJobParameters.COMPATIBILITY_MODE, true)
+    conf.set(CLI.Provenance.PROV, "all");
+    val job = ExtractorJob.getJob(conf)
+    val trainingData = new Path(RuleExtractorTest.trainingDataFile.getAbsolutePath)
+    val filteredData = new Path(RuleExtractorTest.folder.newFile().getAbsolutePath)
+    filterSequenceFile(trainingData, filteredData, conf)
+    FileInputFormat.setInputPaths(job, filteredData)
+    val extractOut = new Path("extractOut")
+    FileOutputFormat.setOutputPath(job, extractOut);
+    job.waitForCompletion(true);
+
+    val s2tOut = new Path("s2t")
+    val s2tJob = (new Source2TargetJob).getJob(conf)
+    FileInputFormat.setInputPaths(s2tJob, extractOut)
+    FileOutputFormat.setOutputPath(s2tJob, s2tOut);
+    val fs2t = Future {
+      s2tJob.waitForCompletion(true)
+    }
+    val t2sOut = new Path("t2s")
+    val t2sJob = (new Target2SourceJob).getJob(conf)
+    FileInputFormat.setInputPaths(t2sJob, extractOut)
+    FileOutputFormat.setOutputPath(t2sJob, t2sOut);
+    val ft2s = Future {
+      t2sJob.waitForCompletion(true)
+    }
+    Await.ready(fs2t, 1 hours)
+    Await.ready(ft2s, 1 hours)
+
+    conf.set(FilterParams.MIN_SOURCE2TARGET_PHRASE, "0.01");
+    conf.set(FilterParams.MIN_TARGET2SOURCE_PHRASE, "1e-10");
```

Comment:
It looks like this is your script for integration testing that takes 5 minutes long after reducing the size of the parallel data. If you want to increase data coverage without hitting the time, you could try modifying FilterParams.MIN_TARGET2SOURCE_PHRASE to 0.01 here (same for FilterParams.MIN_TARGET2SOURCE_RULE). This might reduce considerably the number of extracted rules.